### PR TITLE
Vulcan Module

### PR DESCRIPTION
**Pull request by zmccoy**
This PR addresses: https://github.com/compstak/KafkaStreams4s/issues/36

This PR adds a module for using the Avro serialization library Vulcan (https://github.com/fd4s/vulcan) as an implementation of Codec.  I'm absolutely open to changes that you all may want.  I'd be interested if @vlovgr has any input as well.  

Thank you!

### CODE REVIEW

**Comment by zmccoy** [comment]
@calvinbrown085 Yeah, I did some while in the repl.  I can add a test here as well, just deriving the VulcanSerde and going back and forth using the SerDe.

**Comment by LukaJCB** [comment]
You could add a test using `mapCodec` :)

**Comment by zmccoy** [comment]
@LukaJCB Awesome, I wasn't sure if I was using all of the test suite correctly so that's good to hear!

**Comment by zmccoy** [comment]
Locally I'm not seeing the error that is being seen on the CI: 
==> X compstak.kafkastreams4s.tests.VulcanCodecTest.Using `mapCodec` should serialize and deserialize using the VulcanCodec  0.663s org.apache.kafka.streams.errors.LockException: stream-thread [pool-23-thread-3] task [0_0] Failed to lock the state directory for task 0_0

**Comment by zmccoy** [comment]
I may have a fix locally, it looks like there are issues running multiple suites under the same application_id in the configuration.  Being able to change that per seems to have solved this locally.  I'll make sure and push if that's the case.

**Comment by zmccoy** [comment]
I'm not a big fan of how that changes the api's interactions so I'm up for ideas on how to change to make it better.  Random uuids as the app ids felt a bit odd, but it could be what is used for the properties for those calls that take it in and there would be no change to the api of the runner class.  

I'll need to change the docs as well depending on what you all think on this.  

Thanks!

**Comment by LukaJCB** [comment]
I feel like random uuids would be a great addition since it makes usage less cumbersome. trying to think of a downside but can't really come up with one, wdyt?

**Comment by zmccoy** [comment]
I think that sounds reasonable as well.  So random uuids internally with props so they never get exposed out 👍 .  I'll give that a try and see if there's any issues that I wasn't thinking about.

File: testing/src/main/scala/compstak/kafkastreams4s/testing/KafkaStreamsTestRunner.scala [hunk 1]
```diff
@@ -85,7 +90,7 @@ object KafkaStreamsTestRunner {
 
   def props: ju.Properties = {
     val p = new ju.Properties
-    p.put(StreamsConfig.APPLICATION_ID_CONFIG, "kafkastreams4s-test")
+    p.put(StreamsConfig.APPLICATION_ID_CONFIG, UUID.randomUUID().toString)
```
Comment (LukaJCB) [inline testing/src/main/scala/compstak/kafkastreams4s/testing/KafkaStreamsTestRunner.scala:93]: Can we lift this whole thing into `F` and then flatMap it as needed? :)

Comment (zmccoy) [reply testing/src/main/scala/compstak/kafkastreams4s/testing/KafkaStreamsTestRunner.scala:93]: Yeah totally, doh.

Comment (zmccoy) [reply testing/src/main/scala/compstak/kafkastreams4s/testing/KafkaStreamsTestRunner.scala]: Let me know if https://github.com/compstak/KafkaStreams4s/pull/39/commits/2081792c60dd2736737f92c5189b95966569d05f was what you were thinking.

**Pull request merged**